text/0059-token-metadata-document.md

```diff
@@ -0,0 +1,485 @@
+cannot see the payment in their app, or send it anywhere, or send it back to the Moby user.
+This could lead to disputed payments, bad user experiences, etc.
+
+At the same time, putting token metadata on chain will require an enclave upgrade, while creating
+new tokens on the chain does not. So there is considerable interest in having an off-chain source
+of token metadata that can be validated by clients.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+This proposal creates a source of truth for token metadata for tokens on the MobileCoin blockchain.
+
+Two documents are hosted on the MobileCoin foundation website:
+
+* `https://www.mobilecoin.com/token_metadata.json`: A JSON document containing token metadata
+* `https://www.mobilecoin.com/token_metadata.sig`: An ed25519 signature over the bytes of the .json document. This is binary data.
+
+The ed25519 signing key will be controlled by the mobilecoin foundation.
+This MCIP will be updated, and the public key will be posted here, after the key is created,
+if the proposal is accepted.
+
+Token metadata ed25519 public key hex bytes: `TBD`
+
+The `token_metadata.json` document has the following schema (example):
+
+```
+{
+    signing_timestamp: "10020032320032",
+    metadata: [
+        {
+            token_id: "0",
+            currency_name: "Mobilecoin",
+            ticker_symbol: "MOB",
+            decimals: 12,
+            logo_svg: "b64...==",
+            info_url: "https://www.mobilecoin.com"
+        },
+        {
+            token_id: "1",
+            currency_name: "Electronic Dollars",
+            ticker_symbol: "eUSD",
+            symbol: "$",
+            decimals: 6,
+            logo_svg: "b64...==",
+            info_url: "https://mobilecoin.com/blog/mobilecoin-launches-eusd"
+        },
+        {
+            token_id: "14",
+            currency_name: 'Meowblecoin',
+            ticker_symbol: 'MEOW',
+            decimals: 12,
+            logo_svg: "b64...==",
+            info_url: "https://www.meowblecoin.com"
+        }
+    ],
+}
+```
+
+This metadata list is conceptually a map from `token_id` integers to the metadata
+about the token of that id.
+
+For example:
+
+* A `TxOut` with token id of 0 and a (u64) value of `1.5 * 10^12`, may be displayed as "1.5 MOB" to the user, because the token id corresponds to MOB, and the number of decimals of MOB is 12.
+* A `TxOut` with a token id of 1 and a  (u64) value of `5 * 10^8` may be displayed as `500 eUSD` to the user, because the token id corresponds to eUSD, and the number of decimals is 6.
+* That `TxOut` might also be displayed as `$500` because the symbol of `eUSD` is specified as `$`.
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+The token-metadata json fields have the following semantics:
+
+* `signing_timestamp`: A string representation of the unix timestamp at the time of creating the document's signature.
```

Comment:
That's not really possible. You're going to be editing this file and then at some point later sign it... maybe this should be `last_updated`?

Reply:
:thought_balloon: Since the `.sig` is a separate file could the signing utility update the value during the signing process?